src/super_context.ml

```diff
@@ -10,7 +10,7 @@ module Dir_with_jbuild = struct
     { src_dir : Path.t
     ; ctx_dir : Path.t
     ; stanzas : Stanzas.t
-    ; scope   : Lib_db.Scope.t Lib_db.with_required_by
+    ; scope   : Lib_db.Scope.t With_required_by.t
```

Comment:
I was eyeing at a change like this as well. I'm wondering why is a part of this PR though, I don't quite see the dependency? Can we split this into its own PR or things are a bit too messy.

Reply:
It's because I wanted `With_required_by.reraise`, and then it felt natural to put this type in there. It's true that this PR is big. I can try to split it